units/generate_tasks.go

```diff
@@ -291,7 +291,9 @@ func (j *generateTasksJob) Run(ctx context.Context) {
 		j.AddError(task.MarkGeneratedTasksErr(j.TaskID, err))
 		return
 	}
-	j.AddError(task.MarkGeneratedTasks(j.TaskID))
+	if !shouldNoop {
+		j.AddError(task.MarkGeneratedTasks(j.TaskID))
```

Comment:
What happens in the `shouldNoop && err == nil` case? Is that when another job will be generated and run?

Reply:
That's not a possible scenario. shouldNoop is only true if the error is a duplicate key or results not found error